### ISSUE TITLE
changing axisorder after having added annotations

### ISSUE THREAD

**Issue by k-dominik** [ilastik/ilastik#2234]
## Description

cc @constantinpape  (thanks for bringing this up)

If in pixel classification, one 
* loads data, 
* selects features,
* starts labeling

and then decides that axisorder, or meaning of axes needs to be changed (e.g. `zyx` -> `cyx`)

will result in an error from the following code path:

https://github.com/ilastik/ilastik/blob/60b87a1f02649299989f041ee2428c1d61460b6f/lazyflow/operators/opFeatureMatrixCache.py#L85-L89

or, even worse, you run into something like (which is a bug) (when you start off with changed axes `yxc` -> `yxz` and after labeling go back to `yxc`.

```pytb
  File "/home/dkutra/ilastik-versions/ilastik-1.4.0b1-Linux/ilastik-meta/lazyflow/lazyflow/operators/opReorderAxes.py", line 183, in propagateDirty
    in_roi, self.Input.meta.shape
AssertionError: The dirty ROI (Subregion: start '[0, 0, 0, 0]' stop '[1024, 1344, 3, 1]') appears incorrect for the InputSlot, which has shape (1024, 1344, 1).
```

So, basically, it's not possible to change axis ordering/meaning after one has added annotations.
